specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

```diff
@@ -0,0 +1,609 @@
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      },
+      "type": "oauth2",
+      "description": "Azure Active Directory OAuth2 Flow"
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/usages": {
+      "get": {
+        "tags": [
+          "QuotaInformation"
+        ],
+        "summary": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "description": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "operationId": "Quota_Check",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ProviderIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Quota information will be returned as response with pagination.",
+            "schema": {
+              "$ref": "#/definitions/CurrentQuotaLimits"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests/{requestName}": {
+      "put": {
+        "tags": [
+          "QuotaRequests"
+        ],
+        "summary": "Submits a Quota Request for a resource provider at the specified location.",
+        "description": "Submits Quota change request for a resource provider for the specified location. To use, first make a request to the /usages API to get quota information. This information consists of a list of resources and information regarding those resources. For all the resources in that list which require an update to their quotas, update their limit fields in the response from the /usages to their new values. Then, submit this updated JSON object to this quota request API. This will update the quotas to the values specified.",
+        "operationId": "Quota_Request_Submit",
```

Comment:
Operation IDs are used to generate class+method; i.e. left of first `_` is the class name and to the right is method name.  Also, from the description (and since this is a PUT) it sounds like this API can create a new or update an existing resource?  If so then the operation ID should be `QuotaRequests_CreateOrUpdate`.

Reply:
I can change it to - QuotaRequests_Create. There is no Update operation to QuotaRequest.